keps/498-GroupScheduling/README.md

```diff
@@ -0,0 +1,336 @@
+ 5. Now that the mount point directory path exists in the initContainer, it
+    will exit and allow the main workload container to start.
+
+
+#### Example JobSet spec AFTER webhook injection:
+
+```yaml
+apiVersion: jobset.x-k8s.io/v1alpha2
+kind: JobSet
+metadata:
+  name: my-jobset
+  namespace: default
+spec:
+  # main containers in pods part of `workers` replicatedJob will be blocked
+  # from execution until all pods have started.
+  groupSchedulingConfig:
+    targetReplicatedJobs:
+    - workers
+  replicatedJobs:
+    - name: workers
+      replicas: 3
+      template:
+        spec:
+          backoffLimit: 0
+          completions: 100
+          parallelism: 100
+          template:
+            spec:
+              initContainers:
+                # Check if the started file is present before exiting and starting the main container.
+                - name: group-scheduling-init
+                  image: bash
+                  command:
+                    - bash
+                    - -c
+                    - while true; do if [ -f /mnt/group-scheduling/started ]; then break; fi; sleep 1; done
+                  volumeMounts:
+                    - mountPath: /mnt/group-scheduling
+                      name: group-scheduling
+              containers:
+                - name: pytorch
+                  image: pytorch/pytorch:latest
+                  command:
+                    - bash
+                    - -c
+                    - train.py
+              volumes:
+                - name: group-scheduling
+                  configMap:
+                    # Format: {jobset-name}-started-threshold
+                    name: my-jobset-group-scheduling
+                    optional: true
+```
+
+
+### Test Plan
+
+The testing plan will focus on:
+- Unit and integration tests to validate the webhook injection works as intended
+- E2E tests, since we need pods to be created to injectthe initContainer and ConfigMap volume mount.
+- Scale tests, performance benchmarking
+
+##### Prerequisite testing updates
+
+<!--
+Based on reviewers feedback describe what additional tests need to be added prior
+implementing this enhancement to ensure the enhancements have also solid foundations.
+-->
+
+#### Unit Tests
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+- `controllers`: `04/05/2024` - `25.4%`
+
+#### Integration tests
+-
```

Comment:
We should probably mention some test cases here.

Reply:
Added